### DISCUSSION TITLE
WITH example in sqlpp11

### DISCUSSION THREAD

**Discussion by juandent** [fnc12/sqlite_orm#979] [Ideas]
Hi Eugene,

Found this WITH example in Roland Bock's sqlpp11 for your inspiration:

```C++
auto a = sqlpp::cte(sqlpp::alias::a).as(select(all_of(tab)).from(tab).where(tab.alpha > 3));
for (const auto& row : db(with(a)(select(a.alpha).from(a)).unconditionally()))
{
}
```

**Comment by trueqbit** [comment]
Hi @juandent,

seems like no coincidence that I started working on common table expressions, see https://github.com/fnc12/sqlite_orm/discussions/882.

It still needs some time to ripen, though, and currently I am pushing out PRs of fixes and features that I found while implementing CTEs.

**Comment by juandent** [comment]
this is definitely good news!!
I was unaware of your work here, seems amazing to say the least
I am sure Eugene is excited about this...
Will look at #882 
Good job!
